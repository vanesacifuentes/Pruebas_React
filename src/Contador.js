import React, { Fragment, useState } from 'react'
import PropTypes from 'prop-types'

const Contador = ()=>{

    const [contador, setContador] = useState(0)

    return(
        <Fragment>
           <h1>Este es mi contador:</h1>
           <h2>{contador}</h2>
           <button onClick={() => setContador(contador +1) }> Aumentar </button> 
            <button onClick= {() => setContador(contador -1)}> Restar </button>
            <button onClick= {() => setContador(0) }> Reset </button>
            

        </Fragment>
    )

}

Contador.propTypes={
    contador: PropTypes.number.isRequired

}

export default Contador
