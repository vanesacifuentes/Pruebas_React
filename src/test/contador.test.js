import Contador from '../Contador'
import { shallow } from 'enzyme'



describe('Pruebas sobre el componente contador',()=>{

   test('Probando con enzyme', ()=>{

    const wrapper = shallow(<Contador propNum={0} />)
    expect(wrapper).toMatchSnapshot()

})


  test('Simular click boton aumentar',() => {
    const wrapper = shallow(<Contador propNum={0} />)
    const btn1 = wrapper.find('button').at(0).simulate('click')
    const contando = wrapper.find('h2').text().trim()
    expect(contando).toBe('1')
  
  })


  test('Simular click boton restar',() => {
    const wrapper = shallow(<Contador propNum={0} />)
    const btn1 = wrapper.find('button').at(1).simulate('click')
    const contando = wrapper.find('h2').text().trim()
    expect(contando).toBe('-1')
  
  })

  test('Simular click boton reset',() => {
    const wrapper = shallow(<Contador propNum={0} />)
    const btn1 = wrapper.find('button').at(2).simulate('click')
    const contando = wrapper.find('h2').text().trim()
    expect(contando).toBe('0')
  
  })
  
})