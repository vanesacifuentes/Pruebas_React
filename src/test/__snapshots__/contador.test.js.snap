// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente contador Probando con enzyme 1`] = `
<Fragment>
  <h1>
    Este es mi contador:
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
     Aumentar 
  </button>
  <button
    onClick={[Function]}
  >
     Restar 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
</Fragment>
`;
